app/shop/pages/shop/[shopId].tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  BlitzPage,
+  GetServerSideProps,
+  invokeWithMiddleware,
+  PromiseReturnType,
+  useSession,
+} from 'blitz'
+import { Suspense } from 'react'
+import { FiEdit2 } from 'react-icons/fi'
+
+import { Button } from 'app/core/components/Button'
+import { Spinner } from 'app/core/components/Spinner'
+import { TopBar } from 'app/core/components/TopBar'
+import { setupLayout } from 'app/core/utils/setupLayout'
+import { ShopBio } from 'app/shop/components/shop/ShopBio'
+import { ShopButtons } from 'app/shop/components/shop/ShopButtons'
+import { ShopStats } from 'app/shop/components/shop/ShopStats'
+import getShopProfile from 'app/shop/queries/getShopProfile'
+
+interface ShopProfilePageProps {
+  shop: PromiseReturnType<typeof getShopProfile>
+}
+
+const ShopProfilePage: BlitzPage<ShopProfilePageProps> = ({ shop }) => {
+  return (
+    <div>
+      <Suspense fallback={<Spinner />}>
+        <ShopTopBar shop={shop} />
+        <ShopContainer shop={shop} />
+      </Suspense>
+    </div>
+  )
+}
+
+const ShopTopBar = ({ shop }: ShopProfilePageProps) => {
+  const { userId } = useSession()
+  const isOwner = shop.userId === userId
+  return (
+    <TopBar
+      title={shop.name}
+      actions={
+        isOwner && (
```

Comment:
Shop owner should have a Manage Products button instead

Reply:
to be implemented later

Reply:
closing for now